### ISSUE TITLE
[Accessibility]: Cannot navigate to occurrences using keyboard

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#36587]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34448

### Problem description

It is currently not possible to navigate to occurrences within a file using only keyboard navigation (Tab and Shift + Tab). In contrast, Google code search does support it (from my testing, it works in Chrome but not in Firefox on macOS).

### Expected behavior

It should be possible to navigate to occurrences using keyboard navigation.

### Additional details

This was originally flagged by @umpox in https://github.com/sourcegraph/sourcegraph/issues/33509#issuecomment-1125795436

See that comment for a screen recording and more discussion.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by felixfbecker** [comment]
I wrote an RFC for this a long time about how this could work: [RFC 343 REVIEW: Keyboard navigation for tokens](https://docs.google.com/document/d/1Il-Pa7gGqovuVURhqLTCweOY9u7g-v9zjy5T8y0sPEE/edit#heading=h.trqab8y0kufp)

It should be a lot easier to implement today then back then, with the deprecation of extensions and interactive ranges statically available from GraphQL, and even our tokens now going to definition on a single click (making [Variation: Tokens as links](https://docs.google.com/document/d/1Il-Pa7gGqovuVURhqLTCweOY9u7g-v9zjy5T8y0sPEE/edit#heading=h.asycgaclqrfg) possible).

**Comment by umpox** [comment]
We have plans to fix this in the near future, that will close this ticket.

Closing this in favour of:
- https://github.com/sourcegraph/sourcegraph/issues/42420
- https://github.com/sourcegraph/sourcegraph/issues/42421
